src/app/shared/services/task/task.service.ts

```diff
@@ -3,34 +3,50 @@ import { TemplateFieldService } from "../../components/template/services/templat
 import { AppDataService } from "../data/app-data.service";
 import { arrayToHashmap } from "../../utils";
 import { AsyncServiceBase } from "../asyncService.base";
+import { AppConfigService } from "../app-config/app-config.service";
+import { IAppConfig } from "../../model";
 
 @Injectable({
   providedIn: "root",
 })
 export class TaskService extends AsyncServiceBase {
-  // TODO: These should be set from the deployment/skin level (ultimately should come from templates)
-  highlightedTaskFieldName = "_task_highlighted_group_id";
-  taskGroupsListName = "workshop_tasks";
-
+  // The name of the field that tracks the currently highlighted task group
+  highlightedTaskField: string;
+  // The top-level list of task groups (e.g. modules)
+  taskGroupsListName: string;
   taskGroups: any[] = [];
   taskGroupsHashmap: Record<string, any> = {};
+  tasksFeatureEnabled: boolean;
+
   constructor(
     private templateFieldService: TemplateFieldService,
-    private appDataService: AppDataService
+    private appDataService: AppDataService,
+    private appConfigService: AppConfigService
   ) {
     super("Task");
     this.registerInitFunction(this.initialise);
   }
 
   private async initialise() {
     await this.ensureAsyncServicesReady([this.templateFieldService]);
-    this.ensureSyncServicesReady([this.appDataService]);
-    await this.getListOfTaskGroups();
-    if (this.taskGroups.length > 0) {
-      this.evaluateHighlightedTaskGroup();
+    this.ensureSyncServicesReady([this.appDataService, this.appConfigService]);
+    this.subscribeToAppConfigChanges();
+    if (this.tasksFeatureEnabled) {
+      await this.getListOfTaskGroups();
+      if (this.taskGroups.length > 0) {
+        this.evaluateHighlightedTaskGroup();
+      }
     }
   }
 
+  subscribeToAppConfigChanges() {
```

Comment:
nit(non-blocking) does this need to be a public method? if not better to denote as private and move below public methods